File: Leetcode/C++/Container With Most Water.py
class Solution:
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        left = 0
        right = len(height)-1
        mx = 0

        while left < right:
        	if height[left] > height[right]:
        		mx = max(mx, (right-left)*height[right])
        		right -= 1
        	else:
        		mx = max(mx, (right-left)*height[left])
        		left += 1

        return mx


s = Solution()
height = [1,10, 1, 1000, 10, 1]
print(s.maxArea(height))

File: Leetcode/C++/Container With Most Water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
    	int l=height.size();
    	int mx = 0;
    	int left = 0, right = l-1;

        while(left < right){
        	if(height[left] > height[right]){
        		mx = max(mx, height[right]*(right-left));
        		right --;
        	}
        	else{
        		mx = max(mx, height[left]*(right-left));
        		left ++;
        	}
 
        }

        return mx;
    }
};




// class Solution {
// public:
//     int maxArea(vector<int>& height) {
//     	int l=height.size();
//     	int volume;
//     	int mx = 0;

//         for(int i=0; i<l; i++){
//         	int mx_border = height[l-1]-1;
//         	for(int j=l-1; j>i; j--){
// 	        	if(height[j] > mx_border){
// 	        		mx_border = height[j];
// 	        		volume = (j-i) * min(height[i], mx_border);
// 					mx = max(volume, mx);
// 				}
//         	}
//         }

//         return mx;
//     }
// };